corehq/apps/domain_migration_flags/api.py

```diff
@@ -21,7 +21,7 @@ def set_migration_started(domain, slug):
 
 def set_migration_not_started(domain, slug):
     progress, _ = DomainMigrationProgress.objects.get_or_create(domain=domain, migration_slug=slug)
-    if progress.migration_status == MigrationStatus.IN_PROGRESS:
+    if migration_in_progress(domain, slug, True):
```

Comment:
in addition to this you should make `set_migration_complete` fail if the migration is is in `DRY_RUN` state

Reply:
1356d0b